src/v2/controllers/FormController.ts

```diff
@@ -259,9 +259,10 @@ export default Controller.extend({
       // Because SIW sort of finished its current /transaction/
       sessionStorageHelper.removeStateHandle();
```

Comment:
I think we need to move this down into the `else` conditional and also copy it into the `AutoRedirectView`, because if the user is shown the redirect button (didn't get auto-redirected) but then refreshes the page for some reason, they will get returned to introspect rather than have the state preserved.